clang/test/CodeGenOpenCL/builtins-amdgcn-gfx11.cl

```diff
@@ -49,7 +49,7 @@ void test_s_wait_event_export_ready() {
 }
 
 // CHECK-LABEL: @test_global_add_f32
-// CHECK: {{.*}}call{{.*}} float @llvm.amdgcn.global.atomic.fadd.f32.p1.f32(ptr addrspace(1) %{{.*}}, float %{{.*}})
+// CHECK: = atomicrmw fadd ptr addrspace(1) %addr, float %x syncscope("agent") seq_cst, align 4, !amdgpu.no.fine.grained.memory !{{[0-9]+}}, !amdgpu.ignore.denormal.mode !{{[0-9]+$}}
```

Comment:
why the memory order is seq_cst ? Does this generate the same ISA as before? Can we add some test to emit assembly directly by clang to make sure the ISA does not change?

Reply:
That's the most conservative option. The current intrinsic handling isn't treated as an atomic at all, and the lowering adds a volatile flag instead. With seq_cst you end up with an additional cache flush compared to the current intrinsic. Release seems to be the strongest ordering that doesn't introduce a new flush after 

Running codegen and checking ISA is generally discouraged in clang tests

Reply:
We have all the codegen tests in the backends. Clang tests checking ISA are discouraged. The set of atomic tests is hard enough to maintain as it is

Reply:
can you point to me the codegen tests in backend that make sure the new IR of the builtin generates the same ISA as the old IR of the builtin?

Reply:
Any of the atoimcrmw tests, like llvm/test/CodeGen/AMDGPU/global-atomicrmw-fadd.ll